### DISCUSSION TITLE
Using vector search in Windows

### DISCUSSION THREAD

**Discussion by deepakbkar** [blevesearch/bleve#2124] [Q&A] [answered]
I was able to follow the steps for building `faiss` and use `Bleve` for vector search on Linux.

On Windows, I was able to build `faiss` libraries which produced faiss.dll, faiss.lib and faiss_c.dll.
On building my `main.go` (same code used on Linux), I am getting linker errors. I investigated this and can see that the `faiss_c.dll` does not have these functions exported. 

I wanted to know if anyone has tried using vector search on `Windows 11` and if there is anything I can try to make this work. 
Any suggestion or ideas here will be useful.

Please let me know if you need more information.

Error:
C:/msys64_new/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\abc\AppData\Local\Temp\go-link-917172968\000013.o:index.cgo2.c:(.text+0x3b0): undefined reference to faiss_Index_is_trained' C:/msys64_new/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\Users\abc\AppData\Local\Temp\go-link-917172968\000013.o:index.cgo2.c:(.text+0x410): undefined reference to faiss_Index_merge_from'

**Comment by abhinavdangeti** [comment]
@deepakbkar I believe this has to do with the DLL you've built using faiss. You must make sure you checkout the advertised version of faiss to build the dynamic library - https://github.com/blevesearch/bleve/blob/master/docs/vectors.md#pre-requisites

The code the error calls out is only available in the blevesearch org's fork of faiss.

**Reply by deepakbkar** [reply, marked as answer]
I did use the advertised version. 
I was finally able to make this work by using `-DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON` flag when building `faiss` libraries on Windows. I found this while looking at other C# wrappers created to use `faiss` on Windows.
I am marking this as an answer for the original question.
